docs/src/appendix/deprecated-features.md

```diff
@@ -0,0 +1,49 @@
+# Deprecated Features
+
+This section lists deprecated features.
+
+## Deprecated in 0.23.0-alpha2
+- TOML configuration - deprecated and removed in 0.23.0-alpha2
+- Legacy environment variables - deprecated in 0.23.0-alpha2, to be removed in 0.25.0
```

Comment:
```suggestion
- Legacy environment variables - deprecated in 0.23.0-alpha2, to be removed in 0.24.0
```

As far as I know, we support two minor versions, so when we are on 0.24.x, we will be supporting 0.23.x and 0.24.x - so at that point we can already remove legacy configuration support. Correct me if I'm wrong @menski